gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageClientImplIntegrationTest.java

```diff
@@ -189,9 +213,170 @@ public void uploadViaJournaling() throws IOException {
     writeAndVerifyTemporaryFiles(resourceId, 1);
   }
 
+  @Test
+  public void uploadViaPCUVerifyPartFileCleanup() throws IOException, InterruptedException {
+    String partFilePrefix = name.getMethodName();
+    GoogleCloudStorageOptions storageOptions =
+        GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+            .setWriteChannelOptions(
+                pcuDefaultOptions.toBuilder().setPartFilePrefix(partFilePrefix).build())
+            .build();
+
+    gcs = getGCSImpl(storageOptions);
+    StorageResourceId resourceId = new StorageResourceId(TEST_BUCKET, name.getMethodName());
+
+    writeAndVerifyPartFiles(bufferCapacity, resourceId, partFileCount, 0, partFilePrefix);
+  }
+
+  @Test
+  public void uploadViaPCUVerifyPartFileNotCleanedUp() throws IOException, InterruptedException {
+    String partFilePrefix = name.getMethodName();
+    GoogleCloudStorageOptions storageOptions =
+        GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+            .setWriteChannelOptions(
+                pcuDefaultOptions.toBuilder()
+                    .setPartFilePrefix(partFilePrefix)
+                    .setPartFileCleanupType(PartFileCleanupType.NEVER)
+                    .build())
+            .build();
+
+    gcs = getGCSImpl(storageOptions);
```

Comment:
close this. Also at other places.

Reply:
Code snippet where it is getting closed. Not sure if I understand the concern.
```
  @After
  public void cleanUp() {
    // cleanup any leaked files
    ImmutableSet<String> tempDirs = ImmutableSet.of(GCS_WRITE_TMP_DIR_1, GCS_WRITE_TMP_DIR);
    .
    .
    .

    // close cloudStorage to free up resources
    if (gcs != null) {
      gcs.close();
    }
```